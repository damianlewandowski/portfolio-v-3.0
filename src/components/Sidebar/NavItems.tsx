import React from "react";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import List from "@material-ui/core/List";
import PersonIcon from "@material-ui/icons/Person";
import UpdateIcon from "@material-ui/icons/Update";
import ListAltIcon from "@material-ui/icons/ListAlt";
import { blueGrey, green, purple } from "@material-ui/core/colors";
import EqualizerIcon from "@material-ui/icons/Equalizer";
import { IconItem } from "../../interfaces";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

interface Props {
  onClick?: () => void;
}

interface Item extends IconItem {
  href: string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    resume: {
      color: theme.palette.text.primary,
      fontSize: `${theme.typography.h5.fontSize}`
    },
    link: {
      textDecoration: "none",
      color: "inherit"
    }
  })
);

const NavItems = ({ onClick }: Props) => {
  const classes = useStyles();
  const items: Item[] = [
    {
      text: "Profile",
      icon: <PersonIcon color={"primary"} />,
      href: "#profile"
    },
    {
      text: "Experience",
      icon: <UpdateIcon style={{ color: blueGrey[800] }} />,
      href: "#experience"
    },
    {
      text: "Projects",
      icon: <ListAltIcon style={{ color: purple[500] }} />,
      href: "#projects"
    },
    {
      text: "Skills",
      icon: <EqualizerIcon style={{ color: green[500] }} />,
      href: "#skills"
    }
  ];

  return (
    <List disablePadding={true}>
      <ListItem divider selected>
        <ListItemText
          classes={{ primary: classes.resume }}
          primary={"Resume"}
        />
      </ListItem>
      {items.map(item => (
        <a key={item.text} className={classes.link} href={item.href}>
          <ListItem button onClick={onClick}>
            <ListItemIcon>{item.icon}</ListItemIcon>
            <ListItemText primary={item.text} />
          </ListItem>
        </a>
      ))}
    </List>
  );
};

export default NavItems;
