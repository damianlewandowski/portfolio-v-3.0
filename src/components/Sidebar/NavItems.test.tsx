import React from "react";

import {shallow} from "enzyme";
import NavItems from "./NavItems";

it("renders correctly", () => {
  const wrapper = shallow(<NavItems />);

  expect(wrapper).toMatchSnapshot();
});

