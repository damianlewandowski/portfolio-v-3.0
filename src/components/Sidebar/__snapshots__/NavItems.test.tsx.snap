// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(List))
  disablePadding={true}
>
  <WithStyles(ForwardRef(ListItem))
    divider={true}
    selected={true}
  >
    <WithStyles(ForwardRef(ListItemText))
      classes={
        Object {
          "primary": "makeStyles-resume-1",
        }
      }
      primary="Resume"
    />
  </WithStyles(ForwardRef(ListItem))>
  <a
    className="makeStyles-link-2"
    href="#profile"
    key="Profile"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <PersonIcon
          color="primary"
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Profile"
      />
    </WithStyles(ForwardRef(ListItem))>
  </a>
  <a
    className="makeStyles-link-2"
    href="#experience"
    key="Experience"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <UpdateIcon
          style={
            Object {
              "color": "#37474f",
            }
          }
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Experience"
      />
    </WithStyles(ForwardRef(ListItem))>
  </a>
  <a
    className="makeStyles-link-2"
    href="#skills"
    key="Skills"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <EqualizerIcon
          style={
            Object {
              "color": "#4caf50",
            }
          }
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Skills"
      />
    </WithStyles(ForwardRef(ListItem))>
  </a>
  <a
    className="makeStyles-link-2"
    href="#projects"
    key="Projects"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <ListAltIcon
          style={
            Object {
              "color": "#9c27b0",
            }
          }
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Projects"
      />
    </WithStyles(ForwardRef(ListItem))>
  </a>
</WithStyles(ForwardRef(List))>
`;
